core/dbt/include/global_project/macros/adapters/timestamps.sql

```diff
@@ -0,0 +1,43 @@
+{%- macro current_timestamp() -%}
+    {{ adapter.dispatch('current_timestamp', 'dbt')() }}
+{%- endmacro -%}
+
+{% macro default__current_timestamp() %}
+    current_timestamp
+{% endmacro %}
```

Comment:
### Unblocking this PR
I don't think your change is needed to accomplish the purposes of this back compatibility PR, so let's leave the existing implementation as-is (but you _can_ move it from the old file to its new file) like so:

```suggestion
{% macro default__current_timestamp() -%}
  {{ exceptions.raise_not_implemented(
    'current_timestamp macro not implemented for adapter '+adapter.type()) }}
{%- endmacro %}
```

---
### Providing a path for innovation and change
You proposed change to `current_timestamp` seems like a reasonable one to propose, and you can raise it in a stand-alone PR.